core/src/main/java/org/apache/carbondata/core/util/CarbonThreadFactory.java

```diff
@@ -49,7 +49,7 @@ public CarbonThreadFactory(String name, boolean withTime) {
   @Override public Thread newThread(Runnable r) {
     final Thread thread = defaultFactory.newThread(r);
     if (withTime) {
-      thread.setName(name + "_" + System.currentTimeMillis());
+      thread.setName(name + "_" + System.nanoTime());
```

Comment:
Here it is better to use the generated task number as thread id, so that debug is easier

Reply:
I just made it more accurate by changing from milli to nano.

This is not directly linked with spark task number so cannot use it here, This thread pool is used internally in many places.
I will analyse and handle in other PR if required.